frontend/src/components/dashboard/PremiumOnboarding.tsx

```diff
@@ -26,6 +26,54 @@ export type Props = {
   onPickSubdomain: (subdomain: string) => void;
 };
 
+interface IBrowserContext {
+  extensionsSupported: boolean;
+}
+const BrowserContext = createContext<IBrowserContext>({
+  extensionsSupported: supportsFirefoxExtension() || supportsChromeExtension(),
+});
+interface AddonDescriptionProps {
+  extSupported: boolean;
+  headerMessageId: string;
+  paragraphMessageId: string;
+  linkHref: string;
+  linkMessageId: string;
+}
+
+const _getAddonDescriptionProps = (): AddonDescriptionProps => {
+  if (supportsFirefoxExtension()) {
+    return {
+      extSupported: true,
+      headerMessageId: "multi-part-onboarding-premium-extension-get-title",
+      paragraphMessageId:
+        "multi-part-onboarding-premium-extension-get-description",
+      linkHref:
+        "https://addons.mozilla.org/firefox/addon/private-relay/?utm_source=fx-relay&utm_medium=onboarding&utm_campaign=install-addon",
+      linkMessageId: "multi-part-onboarding-premium-extension-button-download",
+    };
+  }
+  if (supportsChromeExtension()) {
+    return {
+      extSupported: true,
+      headerMessageId:
+        "multi-part-onboarding-premium-chrome-extension-get-title",
+      paragraphMessageId:
+        "multi-part-onboarding-premium-chrome-extension-get-description",
+      linkHref:
+        "https://chrome.google.com/webstore/detail/firefox-relay/lknpoadjjkjcmjhbjpcljdednccbldeb?utm_source=fx-relay&utm_medium=onboarding&utm_campaign=install-addon",
+      linkMessageId:
+        "multi-part-onboarding-premium-chrome-extension-button-download",
+    };
+  }
+  return {
+    extSupported: false,
+    headerMessageId: "",
+    paragraphMessageId: "",
+    linkHref: "",
+    linkMessageId: "",
+  };
+};
+
```

Comment:
I thought about trying to consolidate all of this into EITHER a context OR a helper function. 🤷